### ISSUE TITLE
When "Automatically Scroll Timeline" is on, wrong actual blocks playing are shown...

### ISSUE THREAD

**Issue by Lykdraft** [kmatheussen/radium#1269]
Hey there,

when "Automatically Scroll Timeline" is on,  it seems like the actual sequencer view is off by
2 blocks (in this case block length is 32 lines). When block 1 should be centered below the cursor/playhead
in the sequencer (position 1.1.1) actually the center is around 5.1.1 or 6.1.1. 

(Oh, it depends on window zoom. Wanted to make a video, resized the window to a smaller size.
Now it is off by god-knows how much. Looks like this is very much broken right now. Given that this is 'beta' anyways not sure
if this should be an open ticket anyways...)

**Comment by kmatheussen** [comment]
This function worked right after implementing it, but it's hard to
maintain, and instead of fixing it I marked it "experimental". I'm probably
going to remove it. Is it very useful you think?

**Comment by Lykdraft** [comment]
It is pretty useful because it centers the sequencer timeline view so the overview is quite good
and no need to manually drag the scroll slider around all the time.

**However:** You could consider adding a hotkey to center the timeline manually.
That would do it as well. The view doesn't need to scroll dead-centered all the time, the ability to have it centered
on command does work great too. (I know this from other DAWs, especially in FL Studio this command is very convenient)

**Comment by kmatheussen** [comment]
> **However:** You could consider adding a hotkey to center the timeline manually.
> That would do it as well. The view doesn't need to scroll dead-centered all the time, the ability to have it centered
> on command does work great too. (I know this from other DAWs, especially in FL Studio this command is very convenient)

Doesn't the option "Automatically scroll timeline ..." do this? (I thought it was enabled by default, but maybe it isn't)

**Comment by kmatheussen** [comment]
I mean, isn't that option good enough? (In the "Sequencer" tab in preferences).

**Comment by kmatheussen** [comment]
Here's the hotkey:
```
F10 : ra.evalScheme "(let* ((pos (<ra> :get-song-pos)) (s1 (<ra> :get-sequencer-visible-start-time)) (s2 (<ra> :get-sequencer-visible-end-time)) (d (- s2 s1)) (ns1 (max 0 (round (- pos (/ d 2))))) (ns2 (+ ns1 d))) (<ra> :set-sequencer-visible-start-time ns1) (<ra> :set-sequencer-visible-end-time ns2))"
```

**Comment by Lykdraft** [comment]
No, not really. As it doesn't center the view. It scrolls in "pages".
The thing why people love centered timeline views (I think Ableton was the first to introduce this back in the days)
is that the playhead (and thus the arrangement) is dead-centered. :) Paging is default everywhere, but you are always stuck on the far left or the far right. This is why major DAWs have the "center view on playhead" command.

I mean it really just is a convenience thing. Because there less need to fiddle around with the scroll bar etc.
If it's gone for good also no big deal. But anyways, to get the sequence view up to speed down the line it probably will need some basic things like this in addition to zoom to selection, zoom out to show all, incremental zoom in/out, manual pagescroll left/right and all that jazz...

**Comment by Lykdraft** [comment]
`F10 : ra.evalScheme "(let* ((pos (<ra> :get-song-pos)) (s1 (<ra> :get-sequencer-visible-start-time)) (s2 (<ra> :get-sequencer-visible-end-time)) (d (- s2 s1)) (ns1 (max 0 (round (- pos (/ d 2))))) (ns2 (+ ns1 d))) (<ra> :set-sequencer-visible-start-time ns1) (<ra> :set-sequencer-visible-end-time ns2))"
`

Oh, awesome... Still don't know why I can't get my head around reading scheme. I know how it should be read, but 
it still is awkward to read.. gosh... :)

**Comment by kmatheussen** [comment]
I think the "Automatically scroll timeline..." option keeps the cursor always between 1/4 and 3/4 of the screen, not far left or far right. It's easy to make it narrower the center than that as well if that makes sense.

**Comment by Lykdraft** [comment]
> I think the "Automatically scroll timeline..." option keeps the cursor always between 1/4 and 3/4 of the screen, not far left or far right. It's easy to make it narrower the center than that as well if that makes sense.

Hmmm. Not sure about that. As this is default behaviour in basically all other DAW packages...
I might be better but most likely at some point some people might complain.

**Comment by kmatheussen** [comment]
Hmm, the API for setting sequencer start/end time is not good. It should be only one function for setting both start and end time. If not we have to check beforehand which function to call first to avoid setting illegal values, which is not obvious (as I just demonstrated by not thinking about this when writing the above code). I probably have to change this API.

**Comment by Lykdraft** [comment]
FYI: The "center on playhead" command works. Nice.
